datadog_checks_base/tests/test_time.py

```diff
@@ -0,0 +1,60 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datetime import datetime
+
+import pytest
+from dateutil import tz
+from six import PY2
+
+from datadog_checks.base.utils.time import EPOCH, UTC, get_current_datetime, get_timestamp, normalize_datetime
+
+pytestmark = pytest.mark.time
+
+
+def test_timestamp_type():
+    assert isinstance(get_timestamp(), float)
+
+
+class TestNormalization:
+    def test_replace(self):
+        now = datetime.now()
+        assert now.tzinfo is None
+
+        assert normalize_datetime(now).tzinfo is UTC
+
+    def test_utc(self):
+        nyc = tz.gettz('America/New_York')
+        now = datetime.now(nyc)
+
+        assert normalize_datetime(now).tzinfo is nyc
+
+
+class TestCurrentDatetime:
+    def test_default_utc(self):
```

Comment:
I'm not crazy about tests that rely on `now()`. It might be better to mock out `datetime.now()` and assert that `get_current_datetime()` called with no arguments calls `datetime.now` with `UTC` as the argument?

Alternatively, there's `freezegun`, which works from a practical perspective, but [I've become mildly concerned with keeping it as a dependency](https://github.com/dateutil/dateutil/issues/923).

Reply:
Agree, not fan either relying on current time. +1 for mocking `datetime.now`.

The `assert get_timestamp(now) - get_timestamp(dt) < 0.01` might be **flaky** is the system running the test is slowing down for unrelated reason.

Reply:
Addressed!